utils/doclint/generateApiJson.js

```diff
@@ -92,18 +92,19 @@ function sanitize(result) {
 
 /**
  * @param {Documentation.Type} type
+ * @param {boolean} sortProperties
  */
-function serializeType(type) {
+function serializeType(type, sortProperties = false) {
   /** @type {any} */
   const result = { ...type };
   if (type.properties)
-    result.properties = type.properties.map(serializeProperty);
+    result.properties = (sortProperties ? type.sortedProperties() : type.properties).map(serializeProperty);
   if (type.union)
-    result.union = type.union.map(serializeType);
+    result.union = type.union.map(type => serializeType(type));
```

Comment:
does `.map(serializeType)` not work with default argument?

Reply:
No, because the second argument is the `index` of the map function.